calyx-frontend/src/parser.rs

```diff
@@ -1001,6 +1001,18 @@ impl CalyxParser {
         ))
     }
 
+    fn dynamic_repeat_stmt(input: Node) -> ParseResult<ast::Control> {
```

Comment:
Both the repeat statements return an `ast::Control` and so should have one rule to handle them. Seems like the only difference in the syntax is the `static_word` stuff